File: app/src/main/java/xiaolei/plugintoolbox/utils/SchedulersCompat.java
package xiaolei.plugintoolbox.utils;

import io.reactivex.FlowableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;


public class SchedulersCompat {

    private final static FlowableTransformer ioTransformer = upstream ->
            upstream.subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread());

    public static <T> FlowableTransformer<T, T> applyIoSchedulers() {
        return (FlowableTransformer<T, T>) ioTransformer;
    }
}

File: app/src/main/java/xiaolei/plugintoolbox/utils/SchedulersCompat.kt
package xiaolei.plugintoolbox.utils

import io.reactivex.FlowableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers


object SchedulersCompat {

    private val ioTransformer = FlowableTransformer<Any, Any> { upstream ->
        upstream.subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }

    fun <T> applyIoSchedulers(): FlowableTransformer<T, T> {
        return ioTransformer as FlowableTransformer<T, T>
    }
}
